controller/arbitrage.js

```diff
@@ -552,57 +561,41 @@ class Arbitrage {
 
     }
 
-    async calculateProfit(tx, btcPriceFeed, amount){
+    async calculateProfit(txHash){
         try {
-            if(!amount) {
-                amount = conf.amountArbitrage;
-            }
             console.log("Calculate profit from arbitrage");
-            const receipt = await C.web3.eth.getTransactionReceipt(tx.transactionHash);
-            console.log(receipt);
-
-            if (receipt && receipt.logs) {
-                const logs = abiDecoder.decodeLogs(receipt.logs);
-                const conversionEvent = (logs || []).find(log => log && log.name === "Conversion");
-                console.log(JSON.stringify(logs, null, 2));
-
-                if (conversionEvent && conversionEvent.events) {
-                    const priceFeed = Number(btcPriceFeed)/amount;
-                    console.log(priceFeed)
-                    let {fromToken, toToken, fromAmount, toAmount, trader} = U.parseEventParams(conversionEvent.events);
-                    let toAmountWithPFeed, trade;
-
-                    fromAmount = Number(C.web3.utils.fromWei(fromAmount.toString(), 'ether'));
-                    toAmount = Number(C.web3.utils.fromWei(toAmount.toString(), 'ether'));
-
-                    console.log(fromAmount); console.log(toAmount);
-
-                    if (fromToken.toLowerCase() === conf.testTokenRBTC.toLowerCase()) {
-                        toAmountWithPFeed = Number(fromAmount) * priceFeed;
-                        trade = 'sell btc';
-                    } else {
-                        toAmountWithPFeed = Number(fromAmount) / priceFeed;
-                        trade = 'buy btc';
-                    }
-                    const profit = toAmount - toAmountWithPFeed;
-                    console.log(profit)
-
-                    console.log({trader,
-                        fromToken, toToken,
-                        fromAmount, toAmount,
-                        profit,
-                        trade
-                    })
-
-                    await db.addArbitrage({
-                        adr: trader,
-                        fromToken, toToken,
-                        fromAmount, toAmount,
-                        profit, trade,
-                        txHash: tx.transactionHash
-                    })
-                }
+            let { fromToken, toToken, fromAmount, toAmount, trader } = await this.getArbitrageTxData(txHash);
+
+            let profit, trade;
+            if (fromToken.toLowerCase() === conf.testTokenRBTC.toLowerCase()) {
+                const [pricePriceFeed] = await this.getAmmAndPriceFeedPrices({sourceTokenAddress: fromToken, destTokenAddress: toToken}, toAmount);
+                profit = Number(pricePriceFeed).toFixed(6);
```

Comment:
Shouldn't the profit be the difference between actually received amount and the amount received from price feed?